src/js/view/controls/components/settings/submenu.js

```diff
@@ -12,18 +19,51 @@ export default function SettingsSubmenu(name, categoryButton, isDefault) {
     categoryButtonElement.className += ' jw-submenu-' + name;
     categoryButton.show();
 
-    // return focus to topbar element when tabbing after the first or last item
+    // focus on next or previous element if it exists
+    // if not, focus on last or first element in content items based on index given
+    const focusElement = function(ele, index) {
+        if (ele) {
+            ele.focus();
+        } else if (index !== undefined) {
+            contentItems[index].element().focus();
+        }
+    };
+
     const onFocus = function(evt) {
-        const focus =
-            categoryButtonElement &&
-            evt.keyCode === 9 && (
-                evt.srcElement === contentItems[0].element() && evt.shiftKey ||
-                evt.srcElement === contentItems[contentItems.length - 1].element() && !evt.shiftKey
-            );
+        const nextItem = nextSibling(categoryButtonElement);
+        const prevItem = previousSibling(categoryButtonElement);
+        const nextSubItem = nextSibling(evt.target);
+        const prevSubItem = previousSibling(evt.target);
 
-        if (focus) {
-            categoryButtonElement.focus();
+        switch (evt.keyCode) {
+            case 9: // tab
+                if (evt.shiftKey) {
+                    focusElement(prevItem);
+                } else {
+                    focusElement(nextItem);
+                }
```

Comment:
this could be made a little smaller with a ternary:

`focusElement(evt.shiftKey ? prevItem : nextItem);`